web-share/src/svelte/components/Media.svelte

```diff
@@ -0,0 +1,94 @@
+<!--
+Copyright 2019 Google, Inc.
+
+Licensed under the [Apache License, Version 2.0](LICENSE) (the "License");
+you may not use this file except in compliance with the License. You may
+obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<script>
+	import Button from './Button.svelte';
+
+	export let contentType = '';
+	export let src = '';
+
+	const canShare = 'canShare' in navigator;
+
+	function handleView() {
+		window.location.href = src;
+	}
+
+	async function handleDelete() {
+		const cache = await caches.open('media');
+		await cache.delete(src);
+		window.location.reload();
+	}
+
+	async function handleShare() {
+		const cache = await caches.open('media');
+		const response = await cache.match(src);
+		const blob = await response.blob();
+		const file = new  File([blob], src, {
+			type: response.headers.get('content-type'),
+		});
+		const files = [file];
+		if (canShare && navigator.canShare({files})) {
+			navigator.share({files});
+		}
+	}
+</script>
+
+<style>
+	.card {
+		background-color: var(--tertiary-color);
+		border: 1px solid var(--secondary-color);
+		border-radius: 2px;
+		box-shadow: 1px 1px 1px var(--secondary-color);
+		display: flex;
+		flex-direction: column;
+		margin: 16px;
+		padding: 8px;
+	}
+
+	.buttons {
+		align-self: center;
+		display: flex;
+		justify-content: space-evenly;
+		margin-top: 8px;
+		width: 100%;
+	}
+
+	img, video {
+		height: auto;
+	}
+
+	audio, img, video {
+		max-width: 100%;
+	}
+</style>
+
+<div class="card">
+	{#if contentType.startsWith('image/')}
+		<img {src} alt="An image in the scrapbook.">
+	{:else if contentType.startsWith('video/')}
+		<video {src} controls alt="A video in the scrapbook."></video>
```

Comment:
Issue: When downloading audio and video from media controls, I get "No file" in media notification. Is that intended?

Reply:
If so, you may want to set `controlsList` to `nodownload`: https://developers.google.com/web/updates/2017/03/chrome-58-media-updates#controlslist

Reply:
I noticed it only happens with Google Photos. When "sending file" with the Files app. It works seamlessly.

Reply:
Thanks! I didn't know about `controlsList="nodownload"`. I think adding that makes sense vs. figuring out why some media files will download and others won't.